### PR TITLE
LOAN-147 #comment - Merged LoanParties and LoanGuaranties into LoanCore and MortgageLoans

### PR DESCRIPTION
**Pull request by ElisaKendall**
Signed-off-by: Elisa Kendall <ekendall@thematix.com>

## Description

With this resolution, we've eliminated duplicate terms in LoanParties and LoanGuaranties as well as in LoansAppraisal; merged relevant elements into either LoanCore or MortgageLoans; also further cleaned up LoanCore to eliminate contact details which are now available in FND.

Fixes: #923 / LOAN-147

## Checklist:

- [x] I'm familiar with the [FIBO developer quide](../CONTRIBUTING.md#contributing-to-the-fibo-code). My contribution meets all the requirements described there.
- [x] My contribution follows the [principles of best practices for FIBO](../ONTOLOGY_GUIDE.md).
- [x] My changes have been reconciled with latest master and no merge conflicts remain.
- [x] This PR is related to exactly one issue. The issue is referenced by using a GitHub keyword such as "fixes", "closes", or "resolves".
- [x] Hygiene tests have been applied by a PR with "(WIP)" in title.
- [x] The issue has been tested locally using a reasoner (for ontology changes).

### CODE REVIEW

File: LOAN/LoanTypes/MortgageLoans.rdf [hunk 1]
```diff
@@ -114,16 +114,53 @@
...
+	</owl:Class>
+	
+	<owl:Class rdf:about="&fibo-loan-typ-mtg;MortgageIndemnityInsurancePolicy">
+		<rdfs:subClassOf rdf:resource="&fibo-fbc-dae-gty;InsurancePolicy"/>
+		<rdfs:label xml:lang="en">mortgage indemnity insurance policy</rdfs:label>
+		<skos:definition xml:lang="en">The physical policy for the Mortgage Indemnity Guaranty.</skos:definition>
```
Comment (rivettp) [inline LOAN/LoanTypes/MortgageLoans.rdf]: How is this "physical"? It's not referring to a printed policy document.

File: LOAN/LoanTypes/MortgageLoans.rdf [hunk 2]
```diff
@@ -114,16 +114,53 @@
...
+	</owl:Class>
+	
 	<owl:Class rdf:about="&fibo-loan-typ-mtg;MortgageLender">
```
Comment (rivettp) [inline LOAN/LoanTypes/MortgageLoans.rdf]: Seems suspiciously empty. Is this class even needed?

File: LOAN/LoanTypes/MortgageLoans.rdf [hunk 3]
```diff
@@ -139,13 +176,7 @@
...
 			<owl:Restriction>
 				<owl:onProperty rdf:resource="&fibo-loan-typ-mtg;hasMIGProvider"/>
-				<owl:someValuesFrom rdf:resource="&fibo-loan-loan-gty;LoanMIGProvider"/>
-			</owl:Restriction>
-		</rdfs:subClassOf>
-		<rdfs:subClassOf>
-			<owl:Restriction>
-				<owl:onProperty rdf:resource="&fibo-loan-typ-mtg;hasParty"/>
-				<owl:someValuesFrom rdf:resource="&fibo-loan-loan-pty;LoanServicer"/>
+				<owl:someValuesFrom rdf:resource="&fibo-loan-typ-mtg;LoanMIGProvider"/>
```
Comment (rivettp) [inline LOAN/LoanTypes/MortgageLoans.rdf]: No! Not all loans have a MIG e.g. typically if LTV is less than 80%. 
In any case the loan should link to the MIG policy and then the policy to the provider and other info.

File: LOAN/LoanTypes/MortgageLoans.rdf [hunk 4]
```diff
@@ -227,13 +258,7 @@
 	<owl:ObjectProperty rdf:about="&fibo-loan-typ-mtg;hasMIGProvider">
 		<rdfs:label xml:lang="en">has m i g provider</rdfs:label>
```
Comment (rivettp) [inline LOAN/LoanTypes/MortgageLoans.rdf]: Use MIG not m i g

File: LOAN/LoanTypes/SecuredLoans.rdf [hunk 5]
```diff
@@ -71,14 +71,14 @@
 		<rdfs:subClassOf rdf:resource="&fibo-loan-typ-sec;SecuredLoan"/>
```
Comment (rivettp) [inline LOAN/LoanTypes/SecuredLoans.rdf]: These seem orthogonal. Can someone not guarantee a loan without it also being secured? See https://www.finder.com/guaranteed-loan (3rd heading).

Note also that a guarantee may be *limited* to only part of the amount of the loan.

File: LOAN/LoanTypes/SecuredLoans.rdf [hunk 6]
```diff
@@ -71,14 +71,14 @@
...
-				<owl:someValuesFrom rdf:resource="&fibo-loan-typ-sec;SecuredByGuaranty"/>
+				<owl:onProperty rdf:resource="&fibo-fbc-dae-gty;hasGuarantor"/>
+				<owl:someValuesFrom rdf:resource="&fibo-fbc-dae-gty;Guarantor"/>
 			</owl:Restriction>
 		</rdfs:subClassOf>
 		<rdfs:subClassOf>
 			<owl:Restriction>
-				<owl:onProperty rdf:resource="&fibo-loan-loan-gty;guaranteedBy"/>
-				<owl:someValuesFrom rdf:resource="&fibo-loan-loan-gty;LoanGuarantor"/>
+				<owl:onProperty rdf:resource="&fibo-loan-typ-sec;howSecured.2"/>
```
Comment (rivettp) [inline LOAN/LoanTypes/SecuredLoans.rdf]: Maybe we should have an issue/pass to eliminate any remaining ".n" properties

File: LOAN/LoanContracts/LoanCore.rdf [hunk 7]
```diff
@@ -261,6 +253,14 @@
 		<skos:definition>loan or extension of credit in which the loan principal cannot be increased after funds are dispersed and the loan is partially repaid</skos:definition>
 	</owl:Class>
 	
+	<owl:Class rdf:about="&fibo-loan-ln-ln;Comaker">
```
[the remark this answers is not in the document]
Comment (ElisaKendall) [reply LOAN/LoanContracts/LoanCore.rdf]: fixed

File: LOAN/LoanTypes/MortgageLoans.rdf [hunk 8]
```diff
@@ -114,16 +114,53 @@
 		<rdfs:label xml:lang="en">Investment Mortgage</rdfs:label>
 	</owl:Class>
 	
+	<owl:Class rdf:about="&fibo-loan-typ-mtg;LoanMIGProvider">
+		<rdfs:subClassOf rdf:resource="&fibo-fbc-dae-gty;Guarantor"/>
+		<rdfs:label xml:lang="en">loan m i g provider party</rdfs:label>
```
[the remark this answers is not in the document]
Comment (ElisaKendall) [reply LOAN/LoanTypes/MortgageLoans.rdf]: fixed

File: LOAN/LoanTypes/MortgageLoans.rdf [hunk 9]
```diff
@@ -114,16 +114,53 @@
...
+		<skos:definition xml:lang="en">Mortgage Indemnity Guaranty (MIG) provider. Further notes: SME Review notes 16 Sept: Guaranty - mortgage insurance e.g. insure up to 80% exposure. When you get into indemnification, then for instance if the product doesn&apos;t meet the investor&apos;s requirement such that if it doesn&apos;t get paid then the lender steps in and takes the hit for the loan - this is usually a precondition for securitizing (issuing) the loan in a pool. If the loan is not going to be sold on the secondary market there would be no need to indemnify that loan so this term would not apply. Indemnification is insurance for the investor, while the lender is the one providing that indemnification.</skos:definition>
```
[the remark this answers is not in the document]
Comment (ElisaKendall) [reply LOAN/LoanTypes/MortgageLoans.rdf]: fixed

File: LOAN/LoanTypes/MortgageLoans.rdf [hunk 10]
```diff
@@ -114,16 +114,53 @@
 		<rdfs:label xml:lang="en">Investment Mortgage</rdfs:label>
 	</owl:Class>
 	
+	<owl:Class rdf:about="&fibo-loan-typ-mtg;LoanMIGProvider">
```
[the remark this answers is not in the document]
Comment (ElisaKendall) [reply LOAN/LoanTypes/MortgageLoans.rdf]: fixed

File: LOAN/LoanTypes/MortgageLoans.rdf [hunk 11]
```diff
@@ -114,16 +114,53 @@
...
+		<rdfs:subClassOf rdf:resource="&fibo-fbc-dae-gty;Guarantor"/>
+		<rdfs:label xml:lang="en">loan m i g provider party</rdfs:label>
+		<skos:definition xml:lang="en">Mortgage Indemnity Guaranty (MIG) provider. Further notes: SME Review notes 16 Sept: Guaranty - mortgage insurance e.g. insure up to 80% exposure. When you get into indemnification, then for instance if the product doesn&apos;t meet the investor&apos;s requirement such that if it doesn&apos;t get paid then the lender steps in and takes the hit for the loan - this is usually a precondition for securitizing (issuing) the loan in a pool. If the loan is not going to be sold on the secondary market there would be no need to indemnify that loan so this term would not apply. Indemnification is insurance for the investor, while the lender is the one providing that indemnification.</skos:definition>
+	</owl:Class>
+	
 	<owl:Class rdf:about="&fibo-loan-typ-mtg;MortgageBorrower">
 		<rdfs:label xml:lang="en">mortgage borrower</rdfs:label>
 	</owl:Class>
 	
+	<owl:Class rdf:about="&fibo-loan-typ-mtg;MortgageIndemnityGuaranty">
```
[the remark this answers is not in the document]
Comment (ElisaKendall) [reply LOAN/LoanTypes/MortgageLoans.rdf]: fixed

File: LOAN/LoanTypes/MortgageLoans.rdf (hunk 2 above)
Comment (ElisaKendall) [reply LOAN/LoanTypes/MortgageLoans.rdf]: fixed

File: LOAN/LoanTypes/MortgageLoans.rdf (hunk 3 above)
Comment (ElisaKendall) [reply LOAN/LoanTypes/MortgageLoans.rdf]: fixed - made the restriction min 0; also in the case of having a servicer

File: LOAN/LoanTypes/MortgageLoans.rdf (hunk 4 above)
Comment (ElisaKendall) [reply LOAN/LoanTypes/MortgageLoans.rdf]: removed the property - used hasGuarantor

File: LOAN/LoanTypes/SecuredLoans.rdf (hunk 5 above)
Comment (ElisaKendall) [reply LOAN/LoanTypes/SecuredLoans.rdf]: merged a bit more with LoanCore, which also defined secured loan, but may not have completely addressed this ... should raise another issue to merge this, and unsecured loans into LoanCore

File: LOAN/LoanTypes/SecuredLoans.rdf (hunk 6 above)
Comment (ElisaKendall) [reply LOAN/LoanTypes/SecuredLoans.rdf]: Hoping to do that when secured and unsecured loans are merged into loan core - these seem like fundamental concepts and are partially addressed there as well.   Note that there are still mortgage-specific concepts in LoanCore that need to be teased out / moved to mortgage loans as well.  LoanCore should only include the true core details.

File: LOAN/LoanContracts/LoanCore.rdf (hunk 7 above)
[the remark this answers is not in the document]
Comment (rivettp) [reply LOAN/LoanContracts/LoanCore.rdf]: Puzzled you omitted the hyphen from the URI - that's a legal character.

File: LOAN/LoanContracts/LoanCore.rdf [hunk 12]
```diff
@@ -633,14 +573,14 @@
 		</rdfs:subClassOf>
 		<rdfs:subClassOf>
 			<owl:Restriction>
-				<owl:onProperty rdf:resource="&fibo-loan-ln-ln;hasOriginatorCompany"/>
+				<owl:onProperty rdf:resource="&fibo-loan-ln-ln;hasOriginatingServiceProvider"/>
```
Comment (rivettp) [inline LOAN/LoanContracts/LoanCore.rdf]: I thought we were trying to avoid including the names of range classes in properties. Why not hasOriginator?

File: LOAN/LoanContracts/LoanCore.rdf [hunk 13]
```diff
@@ -633,14 +573,14 @@
...
 				<owl:someValuesFrom>
 					<owl:Class>
 						<owl:intersectionOf rdf:parseType="Collection">
-							<rdf:Description rdf:about="&fibo-fnd-org-fm;FormalOrganization">
+							<rdf:Description rdf:about="&fibo-fbc-pas-fpas;FinancialServiceProvider">
 							</rdf:Description>
 							<owl:Restriction>
-								<owl:onProperty rdf:resource="&lcc-lr;has"/>
+								<owl:onProperty rdf:resource="&lcc-lr;isIdentifiedBy"/>
```
Comment (rivettp) [inline LOAN/LoanContracts/LoanCore.rdf]: Too US-specific

File: LOAN/LoanContracts/LoanCore.rdf [hunk 14]
```diff
@@ -948,23 +882,8 @@
...
-		<rdfs:subClassOf>
-			<owl:Restriction>
-				<owl:onProperty rdf:resource="&fibo-fnd-utl-val;hasDataValue"/>
-				<owl:someValuesFrom rdf:resource="&xsd;string"/>
-			</owl:Restriction>
-		</rdfs:subClassOf>
-		<rdfs:label>phone number</rdfs:label>
-		<skos:definition>a virtual address for communication via phone</skos:definition>
-		<skos:editorialNote>Future consideration: move this property to Addresses ontology (fibo-fnd-plc-adr).</skos:editorialNote>
+		<fibo-fnd-utl-av:explanatoryNote>Similar to a PaymentSchedule except that the dates and amounts are actuals.</fibo-fnd-utl-av:explanatoryNote>
```
Comment (rivettp) [inline LOAN/LoanContracts/LoanCore.rdf]: PaymentHistory should be dated. Especially with hasOutstandingAmount (which should be a property of the Loan not the History.
More practically we should be able easily to turn a Schedule into a History.

File: LOAN/LoanContracts/LoanCore.rdf [hunk 15]
```diff
@@ -1146,35 +1065,36 @@
...
 			<owl:Restriction>
 				<owl:onProperty rdf:resource="&fibo-loan-ln-ln;hasCreditLimit"/>
 				<owl:someValuesFrom rdf:resource="&fibo-fnd-acc-cur;MonetaryAmount"/>
 			</owl:Restriction>
 		</rdfs:subClassOf>
-		<rdfs:label>reverse mortgage loan contract</rdfs:label>
+		<rdfs:label>reverse mortgage</rdfs:label>
 		<skos:definition>a mortgage contract that pays out money to the borrower against a set principal limit that is based on the value of existing equity in the underlying collateral.</skos:definition>
 		<fibo-fnd-utl-av:explanatoryNote>A reverse mortgage and an open end loan both have a credit limit.</fibo-fnd-utl-av:explanatoryNote>
 		<fibo-fnd-utl-av:explanatoryNote>The interest is added to the principal balance.</fibo-fnd-utl-av:explanatoryNote>
```
Comment (rivettp) [inline LOAN/LoanContracts/LoanCore.rdf]: The note does not make much sense from a FIBO point of view.

File: LOAN/LoanContracts/LoanCore.rdf [hunk 16]
```diff
@@ -1798,11 +1720,27 @@ As needed, use the worksFor property to connect appraisers to the appraisal mana
...
+							<rdf:Description rdf:about="&fibo-fnd-agr-ctr;ContractParty">
+							</rdf:Description>
+							<rdf:Description rdf:about="&fibo-fnd-agr-ctr;ContractThirdParty">
+							</rdf:Description>
+						</owl:unionOf>
+					</owl:Class>
+				</owl:intersectionOf>
+			</owl:Class>
+		</rdfs:range>
+		<skos:definition>relates something, typically a loan contract, to the service provider responsible for originated the loan</skos:definition>
```
Comment (rivettp) [inline LOAN/LoanContracts/LoanCore.rdf]: "originated" should be "originating", but it's somewhat circular in leaving "originate" undefined.

File: LOAN/LoanContracts/LoanCore.rdf [hunk 17]
```diff
@@ -1798,11 +1720,27 @@ As needed, use the worksFor property to connect appraisers to the appraisal mana
 		<fibo-fnd-utl-av:adaptedFrom>the 2015 Revised HMDA regulation.</fibo-fnd-utl-av:adaptedFrom>
 	</owl:DatatypeProperty>
 	
-	<owl:ObjectProperty rdf:about="&fibo-loan-ln-ln;hasOriginatorCompany">
-		<rdfs:subPropertyOf rdf:resource="&fibo-fnd-pty-pty;hasParty"/>
-		<rdfs:label>has originator company</rdfs:label>
-		<rdfs:range rdf:resource="&fibo-fnd-org-org;Organization"/>
-		<skos:definition>relates something, typically  a loan contract, to the  company affilliated with the person originating the loan</skos:definition>
+	<owl:ObjectProperty rdf:about="&fibo-loan-ln-ln;hasOriginatingServiceProvider">
```
Comment (rivettp) [inline LOAN/LoanContracts/LoanCore.rdf]: should not include the range in the property name

File: LOAN/LoanTypes/MortgageLoans.rdf [hunk 18]
```diff
@@ -95,16 +104,9 @@
...
 		<rdfs:label xml:lang="en">investment home loan</rdfs:label>
-		<skos:definition xml:lang="en">A loan for a second or subsequent property which is taken out for investment purposes.</skos:definition>
-		<skos:editorialNote xml:lang="en">[formal definition to come from participants] The purpose for this is prospecitve income rather than personal use</skos:editorialNote>
-	</owl:Class>
-	
-	<owl:Class rdf:about="&fibo-loan-typ-mtg;InvestmentHomeLoanContract">
-		<rdfs:subClassOf rdf:resource="&fibo-loan-ln-ln;MortgageLoanContract"/>
-		<rdfs:label xml:lang="en">investment home loan contract</rdfs:label>
-		<owl:disjointWith rdf:resource="&fibo-loan-typ-mtg;SecondHomeLoanContract"/>
+		<owl:disjointWith rdf:resource="&fibo-loan-typ-mtg;SecondHomeLoan"/>
```
Comment (rivettp) [inline LOAN/LoanTypes/MortgageLoans.rdf]: The disjointness does not make sense when the definition includes "second property". Is it not possible for a property to be purchased both for vacation use and as an investment?

File: LOAN/LoanTypes/MortgageLoans.rdf [hunk 19]
```diff
@@ -114,48 +116,43 @@
...
-		</rdfs:subClassOf>
-		<rdfs:subClassOf>
-			<owl:Restriction>
-				<owl:onProperty rdf:resource="&fibo-loan-typ-mtg;hasMIGProvider"/>
-				<owl:someValuesFrom rdf:resource="&fibo-loan-loan-gty;LoanMIGProvider"/>
+				<owl:onProperty rdf:resource="&fibo-fbc-pas-caa;isRealizedBy"/>
+				<owl:someValuesFrom rdf:resource="&fibo-loan-typ-mtg;MortgageIndemnityInsurancePolicy"/>
 			</owl:Restriction>
 		</rdfs:subClassOf>
+		<rdfs:label xml:lang="en">mortgage indemnity guarantee</rdfs:label>
```
Comment (rivettp) [inline LOAN/LoanTypes/MortgageLoans.rdf]: Add abbreviation of MIG

File: LOAN/LoanTypes/MortgageLoans.rdf [hunk 20]
```diff
@@ -195,61 +192,18 @@
 	</owl:Class>
 	
 	<owl:Class rdf:about="&fibo-loan-typ-mtg;SecondHomeLoan">
-		<rdfs:subClassOf rdf:resource="&fibo-loan-typ-mtg;MortgageLoan"/>
+		<rdfs:subClassOf rdf:resource="&fibo-loan-ln-ln;Mortgage"/>
 		<rdfs:label xml:lang="en">second home loan</rdfs:label>
 		<skos:definition xml:lang="en">A loan which may be used for a vacation home or second home used for personal reasons.</skos:definition>
```
Comment (rivettp) [inline LOAN/LoanTypes/MortgageLoans.rdf]: how does this relate to MortgageLoanPurpose?

File: LOAN/LoanTypes/SecuredLoans.rdf [hunk 21]
```diff
@@ -68,17 +75,17 @@
 	</owl:Class>
 	
 	<owl:Class rdf:about="&fibo-loan-typ-sec;GuaranteedLoan">
-		<rdfs:subClassOf rdf:resource="&fibo-loan-typ-sec;SecuredLoan"/>
+		<rdfs:subClassOf rdf:resource="&fibo-loan-ln-ln;SecuredLoan"/>
```
Comment (rivettp) [inline LOAN/LoanTypes/SecuredLoans.rdf]: (thought I made this comment already with a reference)
I think this is wrong: secured and guaranteed are orthogonal.

File: LOAN/LoanContracts/LoanCore.rdf (hunk 7 above)
[the remark this answers is not in the document]
Comment (ElisaKendall) [reply LOAN/LoanContracts/LoanCore.rdf]: For naming conventions, we've been following the typical Semantic Web approach of not including any hyphens, underscores, etc. in names, but include proper punctuation in labels, aside from individuals.

File: LOAN/LoanContracts/LoanCore.rdf (hunk 12 above)
Comment (ElisaKendall) [reply LOAN/LoanContracts/LoanCore.rdf]: In mortgage lending, the originator often refers to the individual banker, so I did not want to change the meaning of what MU had - if you look at the ontology, he distinguished between originating company and person for that reason.  The definitions need work, as do property names, and if we follow some of our other patterns, we could have 3 properties - hasOriginator, hasOriginatingServiceProvider, and hasOriginatingIndividual or something like that - which includes the license.    The terminology here needs work, and should be moved to a US specific ontology, but we'll have to account for the fact that originator often refers to the licensed individual.

File: LOAN/LoanContracts/LoanCore.rdf (hunk 13 above)
Comment (ElisaKendall) [reply LOAN/LoanContracts/LoanCore.rdf]: Yes - for another pass.

File: LOAN/LoanContracts/LoanCore.rdf (hunk 14 above)
Comment (ElisaKendall) [reply LOAN/LoanContracts/LoanCore.rdf]: Yes - for another issue

File: LOAN/LoanContracts/LoanCore.rdf (hunk 15 above)
Comment (ElisaKendall) [reply LOAN/LoanContracts/LoanCore.rdf]: Ok - we need to do several more passes on this in any case
